openedx/features/course_experience/templates/course_experience/course-outline-fragment.html

```diff
@@ -42,25 +42,29 @@
                                     ## Subsection title
                                     <span class="subsection-title">${ subsection['display_name'] }</span>
 
+                                    <%
+                                        # assignment_type could be 'Homework', 'Lab', etc., or empty
+                                        assignment_type = subsection.get('format')
+
+                                        if 'special_exam_info' in subsection:
+                                            assignment_description = None
+                                        else:
+                                            assignment_description = assignment_type
+
+                                        if subsection.get('due') is None:
+                                            due_description = None
+                                        else:
+                                            due_description = _('due {date}')
+
+                                        # Use a ' ' to join the non-empty assignment and due descriptions
+                                        data_string = ' '.join(filter(None, [assignment_description, due_description]))
+                                    %>
+                                    % if assignment_type or 'special_exam_info' in subsection:
                                     <div class="details">
-                                        ## There are behavior differences between rendering of subsections which have
-                                        ## exams (timed, graded, etc) and those that do not.
-                                        ##
-                                        ## Exam subsections expose exam status message field as well as a status icon
-                                        <%
-                                            if subsection.get('due') is None:
```

Comment:
@bjacobel Originally, I just wanted to clean up comments.  However, this if/else was too difficult to follow, and may have had issues.  For example, if we could ever end up with a special exam without a due date, we'd be using the assignment type (`subsection.get('format')`) as the description.

In any case, let me know if you are ok with this refactor.

@dianakhuang: Feel free to review as well.

Thanks.